x-pack/plugins/apm/server/lib/search_strategies/correlations/query_histogram.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+
+import type { ElasticsearchClient } from 'src/core/server';
+
+import { TRANSACTION_DURATION } from '../../../../common/elasticsearch_fieldnames';
+import type {
+  HistogramItem,
+  ResponseHit,
+  SearchServiceParams,
+} from '../../../../common/search_strategies/correlations/types';
+
+import { getQueryWithParams } from './get_query_with_params';
+
+export const getTransactionDurationHistogramRequest = (
+  params: SearchServiceParams,
+  interval: number,
+  fieldName?: string,
+  fieldValue?: string
+): estypes.SearchRequest => {
+  const query = getQueryWithParams(params);
```

Comment:
Should `fieldName` and `fieldValue` perhaps be part of params? Or maybe passed to `getQueryWithParams` as separate args?

```ts
getQueryWithParams({params, fieldName, fieldValue})
```

Reply:
Consolidated here https://github.com/elastic/kibana/pull/99905/commits/7dee7b41cc2dd243dcde7e5a5660c9e5b40532c3